### PR TITLE
Update export of AllowanceCharge fields in TradeLineItem for CII and UBL

### PR DESCRIPTION
**Pull request by Tr1ckMan**
Add fields BasisAmount and Reason of Region SpecifiedTradeAllowanceCharge to XRechnung-Profile because both are elemtens of XRechnung.

### CODE REVIEW

File: ZUGFeRD/InvoiceDescriptor23CIIWriter.cs [hunk 1]
```diff
@@ -504,7 +504,7 @@ public override void Save(InvoiceDescriptor descriptor, Stream stream, ZUGFeRDFo
                             Writer.WriteEndElement();
                             #endregion
 
-                            Writer.WriteOptionalElementString("ram", "Reason", specifiedTradeAllowanceCharge.Reason, Profile.Extended); // not in XRechnung according to CII-SR-128
```
Comment (stephanstapel) [inline ZUGFeRD/InvoiceDescriptor23CIIWriter.cs:507]: same here:
https://github.com/ConnectingEurope/eInvoicing-EN16931/blob/7ce3772aff315588f37e38b509173f253d340e45/cii/schematron/CII/EN16931-CII-syntax.sch#L166

**Review by stephanstapel — COMMENTED** [review]
Thanks for your pull request. The CII-SR-* rules are still existing, so I have no idea how to progress here. I fear that for now we can only merge the optional cbc:BaseAmount

**Comment by stephanstapel** [comment]
The code is now in the repository, I just chose a different path. Thus I'm closing this pull request here, thanks a lot for your contribution!

**Comment by Tr1ckMan** [comment]
It's ok but your solution is not complete for CII export. You missed two things. Field "BasisAmount" and "Reason" must also be written for XRechnung-Profile. I create an PR for you

grafik

**Pull request closed without merging**